runtime/lua/vim/lsp/util.lua

```diff
@@ -269,9 +269,6 @@ function M.convert_input_to_markdown_lines(input, contents)
       end
     end
   end
-  if contents[1] == '' or contents[1] == nil then
```

Comment:
Or maybe something like `if contents[1] == '' or contents[1] == nil and #contents == 1 then` would work?

Reply:
Git blame doesn't necessarily explain why. It might be best to improve the check though as I can imagine the server servers returning an empty doc.